rmw_opensplice_cpp/src/functions.cpp

```diff
@@ -352,7 +352,9 @@ rmw_take(const rmw_subscription_t * subscription, void * ros_message)
   const message_type_support_callbacks_t * callbacks = \
     subscriber_info->callbacks;
 
-  return callbacks->take(topic_reader, ros_message);
+  *taken = callbacks->take(topic_reader, ros_message);
```

Comment:
The code in this file is C++, but the function signatures are in C to comply with the API in `rmw.h`